golos.vesting/golos.vesting.cpp

```diff
@@ -74,9 +74,35 @@ void vesting::on_transfer(name from, name to, asset quantity, std::string memo)
     add_balance(recipient != name() ? recipient : from, converted, has_auth(to) ? to : from);
 }
 
+void vesting::on_transfer_vesting(name from, name to, asset quantity, std::string memo) {
+    eosio_assert(_self != to, "_self == to");
+
+    auto recipient = get_recipient(memo);
+    if (token::get_issuer(config::token_name, quantity.symbol.code()) == from && recipient == name())
+        return;     // just increase token supply
+
+    add_balance(recipient != name() ? recipient : from, quantity, has_auth(to) ? to : from);
+}
+
 void vesting::on_bulk_transfer(name from, std::vector<recipient> recipients) {
+    eosio_assert(recipients.size(), "not item in vector");
+    asset sum_quantity{0, recipients.at(0).quantity.symbol};
     for (auto recipient_obj : recipients)
-        on_transfer(from, recipient_obj.to, recipient_obj.quantity, recipient_obj.memo);
+        sum_quantity += recipient_obj.quantity;
+
+    vesting_table table_vesting(_self, _self.value);    // TODO: use symbol as scope #550
+    auto vesting = table_vesting.find(sum_quantity.symbol.code().raw());
+    eosio_assert(vesting != table_vesting.end(), "Token not found");
+    asset converted = token_to_vesting(sum_quantity, *vesting, -sum_quantity.amount);
+    table_vesting.modify(vesting, name(), [&](auto& item) {
+        item.supply += converted;
```

Comment:
the real converted can differ from this value due rounding.
e.g.: we have 3 recipients and send 3 TOKENs (each recipient receive 1 TOKEN). Price is 1/3, so 3 TOKENs converted to 1 VEST.
`converted == 1 VEST`, but sum of individual conversions are `0.999999 VEST` due rounding